### PR TITLE
upgrade rubocop

### PR DESCRIPTION
**Pull request by ggmichaelgo**
### What are you trying to accomplish?

Currently CI is failing due to Rubocop. ([Failed run](https://github.com/Shopify/liquid-c/runs/4833986435?check_suite_focus=true))

The Rubocop violence is coming from the `tmp` folder which we can exclude from Rubocop.

```bash
RuboCop failed!
Running RuboCop...
Inspecting 26 files
.........................C

Offenses:

tmp/x86_64-linux/liquid_c/3.0.3/.rake-compiler-siteconf.rb:1:1: C: Naming/FileName: The name of this source file (.rake-compiler-siteconf.rb) should use snake_case.
require 'rbconfig'
^
tmp/x86_64-linux/liquid_c/3.0.3/.rake-compiler-siteconf.rb:1:1: C: Style/FrozenStringLiteralComment: Missing frozen string literal comment.
require 'rbconfig'
^
```

### How are you solving this issue?

In this PR, it will add a rule to exclude the `tmp` folder from Rubocop.
This PR will be also upgrading the Rubocop since the Rubocop version `0.93.1` is failing to inspect our rake files.

```
bundle exec rubocop
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Inspecting 25 files
..........An error occurred while Layout/BlockAlignment cop was inspecting /home/michael/liquid-c/rakelib/integration_test.rake:3:0.
To see the complete backtrace run rubocop -d.
..An error occurred while Layout/BlockAlignment cop was inspecting /home/michael/liquid-c/rakelib/rubocop.rake:3:0.
To see the complete backtrace run rubocop -d.
.An error occurred while Layout/BlockAlignment cop was inspecting /home/michael/liquid-c/rakelib/unit_test.rake:3:0.
To see the complete backtrace run rubocop -d.
............

25 files inspected, no offenses detected

3 errors occurred:
An error occurred while Layout/BlockAlignment cop was inspecting /home/michael/liquid-c/rakelib/integration_test.rake:3:0.
An error occurred while Layout/BlockAlignment cop was inspecting /home/michael/liquid-c/rakelib/rubocop.rake:3:0.
An error occurred while Layout/BlockAlignment cop was inspecting /home/michael/liquid-c/rakelib/unit_test.rake:3:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:
0.93.1 (using Parser 3.1.0.0, rubocop-ast 1.15.1, running on ruby 3.1.0 x86_64-linux)
```

### CODE REVIEW

File: liquid-c.gemspec [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # coding: utf-8
 # frozen_string_literal: true
-lib = File.expand_path('../lib', __FILE__)
+
+# rubocop:disable Gemspec/RubyVersionGlobalsUsage
```
Comment (ggmichaelgo) [inline liquid-c.gemspec:4]: [note] we have to disable this Rubocop rule

**Comment by ggmichaelgo** [comment]
> Thank you. Most of the changes are `'` to `"` even without any interpolation. Why that change?

This PR also updates `rubocop-shopify` in order to resolve gem compatibilities, and the new gem forces us to use  
 double quotes ([source](https://github.com/Shopify/ruby-style-guide/blob/v2.4.0/rubocop.yml#L657-L658)):
 ```yml
# ruby-style-guide/rubocop.yml
Style/StringLiterals:
  EnforcedStyle: double_quotes

Style/StringLiteralsInInterpolation:
  EnforcedStyle: double_quotes
```

We could update the Rubcop config for this repo to use single quotes for String Literals - what do you think?

**Comment by tobi** [comment]
ah, didn't know that. :+1:

**Review by macournoyer — APPROVED** [review]
Thanks for taking care of this Michael! 🔥 

Here's the reasoning behind our new double quote rule (it's for consistency) https://github.com/Shopify/ruby-style-guide/pull/238. Makes a lot of sense.

**Pull request merged**